File: python-client-sdk/src/sdk/models/shared/codat_datacontracts_datasets_paymentpagedresponselinksmodel.py
from __future__ import annotations
import dataclasses
from ..shared import codat_datacontracts_datasets_paymentpagedresponsehrefmodel as shared_codat_datacontracts_datasets_paymentpagedresponsehrefmodel
from dataclasses_json import Undefined, dataclass_json
from sdk import utils
from typing import Optional


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CodatDataContractsDatasetsPaymentPagedResponseLinksModel:
    current: Optional[shared_codat_datacontracts_datasets_paymentpagedresponsehrefmodel.CodatDataContractsDatasetsPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('current'), 'exclude': lambda f: f is None }})
    next: Optional[shared_codat_datacontracts_datasets_paymentpagedresponsehrefmodel.CodatDataContractsDatasetsPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next'), 'exclude': lambda f: f is None }})
    previous: Optional[shared_codat_datacontracts_datasets_paymentpagedresponsehrefmodel.CodatDataContractsDatasetsPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous'), 'exclude': lambda f: f is None }})
    self: Optional[shared_codat_datacontracts_datasets_paymentpagedresponsehrefmodel.CodatDataContractsDatasetsPaymentPagedResponseHrefModel] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('self'), 'exclude': lambda f: f is None }})
    

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetspaymentpagedresponselinksmodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsPaymentPagedResponseHrefModel } from "./codatdatacontractsdatasetspaymentpagedresponsehrefmodel";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsPaymentPagedResponseLinksModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "current" })
  @Type(() => CodatDataContractsDatasetsPaymentPagedResponseHrefModel)
  current?: CodatDataContractsDatasetsPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "next" })
  @Type(() => CodatDataContractsDatasetsPaymentPagedResponseHrefModel)
  next?: CodatDataContractsDatasetsPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "previous" })
  @Type(() => CodatDataContractsDatasetsPaymentPagedResponseHrefModel)
  previous?: CodatDataContractsDatasetsPaymentPagedResponseHrefModel;

  @SpeakeasyMetadata()
  @Expose({ name: "self" })
  @Type(() => CodatDataContractsDatasetsPaymentPagedResponseHrefModel)
  self?: CodatDataContractsDatasetsPaymentPagedResponseHrefModel;
}